### ISSUE TITLE
typing `document.b' in an empty document kills the server

### ISSUE THREAD

**Issue by cbourjau** [ananthakumaran/tide#7]
Hello,
sorry, I am very new to ts and thus also to tide, so I am sorry if I am missing something obvious. However, I discovered that the following minimum non-working example crashes tide and requires a restart of the server:
`tsconfig.json`:
```javascript
{
    "compilerOptions": {
        "target": "ES5",
        "module": "commonjs",
        "sourceMap": true,
	"rootDir": ".",
        "emitDecoratorMetadata": true,
        "experimentalDecorators": true
    },
    "exclude": [
        "node_modules"
    ]
}
```
Creating a new .ts file in the same folder works initially. Ie. autocompletion and type base method completion works as expected. However, just typing `document.b` (I was going for `document.body`) kills the server and fills the `*Message*` buffer with the following:
```
error in process filter: json-read-object: Bad JSON object: ":", 115
error in process filter: Bad JSON object: ":", 115
error in process filter: json-read-object: Bad JSON object: ":", 115
error in process filter: Bad JSON object: ":", 115
```
I was able to reproduce this behavior with node v4.2.2 and v5.1.0 and tide from elpa 20151015.1820

Any ideas on the reason for this behavior and how to solve the problem?

**Comment by ananthakumaran** [comment]
Are you on OSX? https://github.com/Microsoft/TypeScript/issues/2758 might be the cause. It used to happen for node version < 0.12.x (That's the reason I ask to install node >= 0.12.x in readme). I am on OSX and node v4.1.0, I don't experience the problem. So it is likely got reintroduced again in 4.2.x

**Comment by cbourjau** [comment]
Thanks for the fast feedback! Much appreciated! I am on Ubuntu 15.04 installing node via nvm. I just tried v4.1.0 and unfortunately am seeing the same problem. Any ideas on how to debug this or what the reason could be? I saw the issue you linked but am not sure whether it is really applicable.

**Comment by ananthakumaran** [comment]
tide uses json-read-object in only one [place](https://github.com/ananthakumaran/tide/blob/master/tide.el#L291), to decode the response from the `tsserver`. So it is very likely the same issue I mentioned above.

```lisp
(setq tide-tsserver-process-environment '("TSS_LOG=-level verbose"))
```
This would create a log file in path-to-tide/tsserver/.logxxx . You can also add `  (message "%S" response)` in https://github.com/ananthakumaran/tide/blob/master/tide.el#L183

**Comment by ananthakumaran** [comment]
@kraf You could try to backport the fix (https://github.com/Microsoft/TypeScript/pull/5408) to 1.6

**Comment by ananthakumaran** [comment]
I just realized that they have released 1.7. I will try to update tsserver to 1.7, which should hopefully fix the problem.

**Comment by ananthakumaran** [comment]
Updated master to 1.7.3. Could you check whether it fixes your problem.

**Comment by kraf** [comment]
Works for me! Thanks for updating it so quickly, this is awesome!

**Comment by wamei** [comment]
I had the same problem on Debian 6.0.10.
but this update fixed my problem. Many thanks!

**Comment by cbourjau** [comment]
Indeed, my original problem is fixed using the latest tide and typescript 1.7.3 on npm 5.1.0. Many thanks for this fantastic package and I take the liberty of marking this closed(?)
